### PR TITLE
PEER-734: Switch UI from the Graph new index

### PR DESCRIPTION
**Pull request by andreilarin**

### CODE REVIEW

File: app/utils/mesh.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(props),
+  });
+
+  return response.json();
+}
+
+export const getUserDataFromMesh = (item) => {
+  const user = { ...item };
```
Comment (Lampardks) [inline app/utils/mesh.js:15]: delete

File: app/utils/mesh.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  const ratings = user.usercommunityrating;
+  const achievements = user.userachievement.map(({ achievementId }) => ({
+    id: achievementId,
+  }));
+  delete user.userachievement;
+  delete user.usercommunityrating;
```
Comment (Lampardks) [inline app/utils/mesh.js:21]: don't use delete

```
const {userachievement, usercommunityrating, ...user} = item;
```

File: app/utils/mesh.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    achievements,
+    ratings,
+  };
+};
+
+const getCommentDataFromMesh = (item) => {
+  const comment = { ...item };
+  const author = getUserDataFromMesh(item.user[0]);
+
+  delete comment.user;
```
Comment (Lampardks) [inline app/utils/mesh.js:33]: don't use delete

File: app/utils/mesh.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  return { ...item, author };
+};
+
+const getReplyDataFromMesh = (item, replyComments) => {
+  const reply = { ...item };
```
Comment (Lampardks) [inline app/utils/mesh.js:38]: delete

File: app/utils/mesh.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  const comments = replyComments
+    .filter(
+      (comment) => comment.parentReplyId === Number(reply.id.split('-')[1]),
+    )
+    .map((comment) => getCommentDataFromMesh(comment));
+  const author = getUserDataFromMesh(reply.user[0]);
+
+  delete reply.comment;
+  delete reply.user;
```
Comment (Lampardks) [inline app/utils/mesh.js:47]: don't use delete

File: app/utils/mesh.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  return {
+    ...reply,
+    author,
+    comments,
+  };
+};
+
+export const getPostDataFromMesh = (item) => {
+  const post = { ...item };
```
Comment (Lampardks) [inline app/utils/mesh.js:56]: delete

File: app/utils/mesh.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+  const author = getUserDataFromMesh(post.user[0]);
+  const comments = post.comment
+    .filter((comment) => comment.parentReplyId === 0)
+    .map((comment) => getCommentDataFromMesh(comment));
+
+  delete post.posttag;
+  delete post.reply;
+  delete post.comment;
+  delete post.user;
```
Comment (Lampardks) [inline app/utils/mesh.js:70]: delete

File: app/utils/mesh.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+    author,
+    replies,
+    comments,
+  };
+};
+
+export const renameRepliesToAnswers = (post) => {
+  const result = { ...post, answers: post.replies };
+  delete result.replies;
+  return result;
```
Comment (Lampardks) [inline app/utils/mesh.js:83]:
```
const {replies, ...rest} = post;

return {
...post,
answers: replies
}
```

File: app/utils/mesh.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+};
+
+export const getHistoryDataFromMesh = (item) => {
+  const history = { ...item };
```
Comment (Lampardks) [inline app/utils/mesh.js:87]: delete

File: app/utils/mesh.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+  const reply = history.replyId ? { id: history.replyId } : undefined;
+  const comment = history.commentId ? { id: history.commentId } : undefined;
+
+  delete history.replyId;
+  delete history.commentId;
```
Comment (Lampardks) [inline app/utils/mesh.js:92]: don't use delete

**Review by alexeishulga — APPROVED** [review]

**Pull request closed without merging**